### ISSUE TITLE
Isolated projects feature doesn't work well with maven publish

### ISSUE THREAD

**Issue by dkrasnoff** [gradle/gradle#30186]
### The whole project example
The whole project you can find here: https://github.com/dkrasnoff/simple-publish

### Description
We have two subprojects with `java-library` and `maven-publish` plugins.
Project `app` has api dependency on project `lib` and custom publication such configuration:
```kotlin
plugins {
    id("java-library")
    `maven-publish`
}
...
dependencies {
    api(project(":lib"))
}

publishing {
    publications.create<MavenPublication>("maven") {
        from(components.findByName("java"))
        artifactId = "app-publication"
    }
}
```

Project `lib` has a custom publication:
```kotlin
plugins {
    id("java-library")
    `maven-publish`
}
...
publishing {
    publications.create<MavenPublication>("maven") {
        from(components.findByName("java"))
        artifactId = "custom-lib"
    }
}
```

In `gradle.properties` set:
```
org.gradle.unsafe.isolated-projects=true
```

Then we run:

```
./gradlew :app:publishMavenPublicationToMavenLocal
```

And the build finishes with such error:
```
Project ':app' cannot access 'Project.group' functionality on another project ':lib'
```

### Expected Behavior

When we call `:app:publishMavenPublicationToMavenLocal` with `org.gradle.unsafe.isolated-projects=true` the build is successful, as it is with `org.gradle.unsafe.isolated-projects=false`

### Context (optional)

In Kotlin Gradle plugin we are trying to support the isolated project feature for Kotlin multiplatform publication and trying to base our behavior on Gradle mechanisms, but they are not isolated project compatible as well

### Steps to Reproduce

- download the example project: https://github.com/dkrasnoff/simple-publish
- run `./gradlew :app:publishMavenPublicationToMavenLocal`

### Gradle version

8.9

### Build scan URL (optional)

https://scans.gradle.com/s/od7xp6m73whv6

**Comment by ov7a** [comment]
The issue is in the backlog of the relevant team and is prioritized by them.

**Comment by Goooler** [comment]
Yeah, I suffered the related `Project.group` and `Project.version` issues in Shadow, see https://github.com/GradleUp/shadow/pull/1139#issuecomment-2582235468.
